Source/WebKit/UIProcess/API/glib/WebKitDownloadClient.cpp

```diff
@@ -93,28 +93,30 @@ class DownloadClient final : public API::DownloadClient {
 
     void didFail(DownloadProxy& downloadProxy, const ResourceError& error, API::Data*) override
     {
-        if (webkitDownloadIsCancelled(m_download.get()))
+        if (!m_download)
             return;
 
-        ASSERT(m_download);
-        webkitDownloadFailed(m_download.get(), error);
+        if (webkitDownloadIsCancelled(m_download.get())) {
+            // Cancellation takes precedence over other errors.
+            webkitDownloadCancelled(m_download.get());
```

Comment:
This particular call looks unnecessary as related to this change.  didFinish is the only place that needs it because of this change.

Reply:
The original issue was if a download was fast enough `didFinish` happened before the cancel callback. So while I don't have a test case I think it is plausible `didFail` also could be faster.